### PR TITLE
Revert the use of _ to separate the path element in keys for the Ax sweeper plugin

### LINKED ISSUE #913: [Ax-Plugin] Switch back to the usage of "." in place of "_"
> > I abandoned the idea of compact keys in OmegaConf. This is a good chance to switch back to the more intuitive usage of dot and not _ to separate the path element in keys:
> > ```yaml
> >       params:
> >         banana.x:
> >           type: range
> >           bounds: [-5, 5]
> > ```

_Originally posted by @omry in https://github.com/facebookresearch/hydra/pull/912#issuecomment-679236669_

### PR DESCRIPTION
**Pull request by shagunsodhani**
Fixes #913.

## Motivation

Revert the use of _ to separate the path element in keys for the Ax

### Have you read the [Contributing Guidelines on pull requests](https://github.com/facebookresearch/hydra/blob/master/CONTRIBUTING.md)?

Yes

## Test Plan

CI

## Related Issues and PRs

#913

### CODE REVIEW

**Comment by omry** [comment]
no documentation updates?

File: plugins/hydra_ax_sweeper/tests/config/params/nested_with_escape_char.yaml [hunk 1]
```diff
@@ -1,8 +1,8 @@
 # @package hydra.sweeper.ax_config.params
-quadratic_x\_arg:
+quadratic.x_arg:
```
Comment (omry) [inline plugins/hydra_ax_sweeper/tests/config/params/nested_with_escape_char.yaml:2]: Should this be deleted? the file name no longer represent what this is doing.

**Review by omry — CHANGES_REQUESTED** [review]

**Comment by omry** [comment]
Try to be more proactive, you should have just as much context on this change as me, probably even more.

File: plugins/hydra_ax_sweeper/tests/config/params/nested_with_escape_char.yaml (hunk 1 above)
Comment (shagunsodhani) [reply plugins/hydra_ax_sweeper/tests/config/params/nested_with_escape_char.yaml:2]: Ouch my bad. Thanks for spotting.

**Review by omry — APPROVED** [review]
lgtm

**Pull request merged**